File: exptr_api/internal/models/categories.go
package models

import (
	"time"

	"alex_gorbunov_exptr_api/internal/domain"
	"alex_gorbunov_exptr_api/internal/lib/api/response"
)

type CategoryRequest struct {
	UserID    string    `json:"user_id"`
	Name      string    `json:"name"`
	Type      string    `json:"type"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
	Color     string    `json:"color"`
	Icon      string    `json:"icon"`
}

type CategoryResponse struct {
	response.Response
}

type GetCategoriesResponse struct {
	response.Response
	Categories []domain.Category `json:"categories"`
}

File: exptr_api_p/exptr_api_p/entities/categories.py
import datetime

from pydantic import BaseModel

from exptr_api_p.entities import response


class Category(BaseModel):
    id: int
    user_id: int
    name: str
    type: str
    created_at: datetime.datetime
    updated_at: datetime.datetime
    color: str
    icon: str
    

class CategoryRequest(Category):
    user_id: int
    name: str
    type: str
    created_at: datetime.datetime
    updated_at: datetime.datetime
    color: str
    icon: str

class CategoryResponse(BaseModel):
    response: response.Response
    
class GetCategoriesResponse(BaseModel):
    response: response.Response
    categories: list[Category]